membership_prorate/models/account_invoice.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2015 Tecnativa - Pedro M. Baeza
+# Copyright 2017-18 Tecnativa - David Vidal
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from datetime import timedelta
+
+from odoo import api, fields, models
+from odoo.tests import Form
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _get_membership_interval(self, product, date):
+        """Get the interval to evaluate as the theoretical membership period.
+        :param product: Product that defines the membership
+        :param date: date object for the requested date (if needed for
+        inherited computations)
+        :return: A tuple with 2 date objects with the beginning and the
+        end of the period
+        """
+        date_from = product.membership_date_from
+        date_to = product.membership_date_to
+        return date_from, date_to
+
+    def _prepare_invoice_line_prorate_vals(self, invoice_line):
+        product = invoice_line.product_id
+        date_invoice = invoice_line.move_id.invoice_date or fields.Date.today()
+        date_from, date_to = self._get_membership_interval(product, date_invoice)
+        if date_invoice < date_from:
```

Comment:
@edlopen, basically, for your exact scenario special module is there [membership_prorate_variable_period](https://github.com/OCA/vertical-association/tree/14.0/membership_prorate_variable_period). which is a bridge module (auto install True) for membership_prorate and membership_variable_period modules. which is solving the problem 
so when the membership_prorate_variable_period module is migrated, it will solve your issue.